axelrod/strategies/grudger.py

```diff
@@ -293,16 +294,18 @@ def strategy(self, opponent: Player) -> Action:
         The punishment is in the form of 'd' defections followed by a penance of
         'c' consecutive cooperations.
         """
```

Comment:
This startegy was changed to fit the edge cases where n=0 or d=0 or c=0.  see the tests here for an example [get url to line 322 of test].  If the edge cases are invalid, then this change is totally unnecessary.